sdk/confidentialledger/Azure.Security.CodeTransparency/src/Receipt/DidWebReference.cs

```diff
@@ -0,0 +1,170 @@
+        {
+            if (issuer == null)
+                throw new Exception("invalid did:web issuer: null");
+            string[] parts = issuer.Split(':', (char)StringSplitOptions.RemoveEmptyEntries);
+            if (parts.Length < 3)
+                throw new Exception("invalid did:web issuer: " + issuer);
+
+            string prefix = parts[0];
+            string method = parts[1];
+            string location = parts[2];
+            if (prefix != "did" || method != "web" || location.Trim().Length == 0)
+                throw new Exception("invalid did:web issuer: " + issuer);
+
+            string fqdn = HttpUtility.UrlDecode(location);
+
+            string pathToDidDoc;
+            if (parts.Length == 3)
+                pathToDidDoc = DID_WEB_DOC_WELLKNOWN_PATH;
+            else
+                pathToDidDoc = string.Join("/", parts.ToList().Skip(3).ToArray());
+
+            StringBuilder sbUrl = new();
+            sbUrl.Append("https://");
+            sbUrl.Append(fqdn);
+            sbUrl.Append('/');
+            sbUrl.Append(pathToDidDoc);
+            sbUrl.Append('/');
+            sbUrl.Append(DID_WEB_DOC_NAME);
+            return new Uri(sbUrl.ToString());
+        }
+
+        /// <summary>
+        /// Extract and parse service certificate from the did web doc.
+        /// </summary>
+        /// <param name="doc">DidDocument object.</param>
+        /// <param name="keyId">Key id to map the right service certificate from the did web doc</param>
+        /// <returns>Service certificate.</returns>
+        public static X509Certificate2 ParseCert(DidDocument doc, string keyId)
+        {
+            X509Certificate2 cert = null;
+            if (!keyId.StartsWith("#"))
+                keyId = "#" + keyId;
+            string expectedMethodKey = doc.Id + keyId;
+            foreach (DidDocumentKey method in SupportedAssertionMethods(doc))
+            {
+                if (method.Id == expectedMethodKey)
+                {
+                    cert = JwkToCert(method);
+                    break;
+                }
+            }
+            return cert ?? throw new Exception($"No matching assertion method found for key id {{{keyId}}} in doc {{{doc}}}");
+        }
+
+        /// <summary>
+        /// Identify supported JwkAssertionMethod to get the right service certificate
+        /// </summary>
+        /// <param name="doc">DidDocument object</param>
+        /// <returns>list of supported DidDocumentKey to get the right service certificate.</returns>
+        public static List<DidDocumentKey> SupportedAssertionMethods(DidDocument doc)
+        {
+            List<DidDocumentKey> methods = new();
+            foreach (DidDocumentKey method in doc.AssertionMethod)
+            {
+                if (method.Type == SUPPORTED_ASSERTION_METHOD_TYPE)
+                    methods.Add(method);
+            }
+            return methods;
+        }
+
+        /// <summary>
```

Comment:
I'd strongly suggest doing a pass over all the doc comments.  These are published to MS Docs as the official API documentation.  There are many here which I don't believe would be helpful to customers looking to understand how the library works.